src/backend/distributed/commands/database.c

```diff
@@ -688,6 +695,45 @@ PreprocessDropDatabaseStmt(Node *node, const char *queryString,
 }
 
 
+List *
+PreprocessAlterDatabaseOwnerStmt(Node *node, const char *queryString,
+								 ProcessUtilityContext processUtilityContext)
+{
+	const DistributeObjectOps *ops = GetDistributeObjectOps(node);
+	Assert(ops != NULL);
+
+	if (ops->featureFlag && *ops->featureFlag == false)
+	{
+		/* not propagating when a configured feature flag is turned off by the user */
+		return NIL;
+	}
```

Comment:
given that we're implementing a custom Preprocess function for this DDL statement, let's directly check EnableAlterDatabaseOwner here instead of providing it via dist_object_ops.c

```suggestion
	if (!EnableAlterDatabaseOwner)
	{
		return NIL;
	}
```